internal/hcp/registry/metadata/os.go

```diff
@@ -0,0 +1,132 @@
+package metadata
+
+import (
+	"log"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+)
+
+type OSInfo struct {
+	Name    string
+	Arch    string
+	Version string
+}
+
+// CommandExecutor is an interface for executing commands.
+type CommandExecutor interface {
+	Exec(name string, arg ...string) ([]byte, error)
+}
+
+// DefaultExecutor is the default implementation of CommandExecutor.
+type DefaultExecutor struct{}
+
+// Exec executes a command and returns the combined output.
+func (d DefaultExecutor) Exec(name string, arg ...string) ([]byte, error) {
+	cmd := exec.Command(name, arg...)
+	return cmd.CombinedOutput()
+}
+
+var executor CommandExecutor = DefaultExecutor{}
```

Comment:
Is there a possibility of introducing another command executor in the future? It seems like the default executor is so far the only thing the code needs. It would be more simple and easier to read the code without the interface in the middle if that's only used by one executor.

Reply:
Well as you said in the comment below it is important to do it this way to make the testing easier. I got the idea from [this](https://stackoverflow.com/questions/71102318/how-to-mock-exec-cmd-exec-command) stackoverflow post.